lib/tapioca/dsl/compilers/active_record_associations.rb

```diff
@@ -316,6 +317,32 @@ def declaration(reflection)
           end
         end
 
+        sig { params(reflection: ReflectionType).returns(T.nilable(String)) }
+        def association_guide(reflection)
+          association = case reflection
+          when ActiveRecord::Reflection::HasOneReflection
+            "the-has-one-association"
+          when ActiveRecord::Reflection::HasManyReflection
+            "the-has-many-association"
+          when ActiveRecord::Reflection::HasAndBelongsToManyReflection
+            "the-has-and-belongs-to-many-association"
+          when ActiveRecord::Reflection::BelongsToReflection
+            "the-belongs-to-association"
+          when ActiveRecord::Reflection::ThroughReflection
+            delegate_reflection = reflection.send(:delegate_reflection)
+            declaration = declaration(delegate_reflection)
+            if declaration == "has_one"
+              "the-has-one-through-association"
+            else
+              "the-has-many-through-association"
+            end
+          end
+
+          if association
+            "🔗 [Rails Guides](https://guides.rubyonrails.org/association_basics.html##{association})"
```

Comment:
I would really like the comment that we add to have more informational content. The link is nice, but on its own, not that helpful to the reader. For example, what does `Rails Guides` mean? Guides for what exactly?

I would prefer to have a longer comment that says something like:
```
This method is created at runtime by ActiveRecord on the `Post` class since `Post` has declared a `has_one` association to `User`.

🔗 [Rails guide for `has_one` associations](https://guides.rubyonrails.org/association_basics.html##{association})
```